### ISSUE TITLE
RequiredSvVoltage

### ISSUE THREAD

**Issue by VeikoAunapuu** [Baltic-RCC/EMF#127]

**Comment by Haigutus** [comment]
Possible fix

```python
    # Fix missing injection voltages
    # Injections <- Terminal -> Topological Nodel <- Voltage
    injections = solved_data.type_tableview('EquivalentInjection').reset_index()
    injections = injections.merge(solved_data.type_tableview('Terminal').reset_index(), left_on="ID", right_on="Terminal.ConductingEquipment", suffixes=("_Injection", "_Terminal"))
    injections = injections.merge(solved_data.type_tableview('SvVoltage'), left_on="Terminal.TopologicalNode", right_on="SvVoltage.TopologicalNode")
    injections = injections.merge(solved_data.type_tableview('SvPowerFlow'), left_on="ID_Terminal", right_on="SvPowerFlow.Terminal")
    injections[['ID_Injection', 'EquivalentInjection.p', 'EquivalentInjection.q', 'ACDCTerminal.connected', 'SvVoltage.angle', 'SvVoltage.v', 'SvPowerFlow.p', 'SvPowerFlow.q']]

**Comment by mr0321** [comment]
if #125 is solved then pypowsybl ver 1.6.0 can solve this issue (maybe need to check duplicate values)

**Comment by mr0321** [comment]
In newer version it iterates over the dangling lines to get the voltage values
```
private static void writeVoltagesForBoundaryNodes(Network network, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
        for (DanglingLine dl : network.getDanglingLines(DanglingLineFilter.ALL)) {
            Bus b = dl.getTerminal().getBusView().getBus();
            String topologicalNode = dl.getProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TOPOLOGICAL_NODE_BOUNDARY);
            if (topologicalNode != null) {
                if (dl.hasProperty("v") && dl.hasProperty("angle")) {
                    writeVoltage(topologicalNode, Double.parseDouble(dl.getProperty("v", "NaN")), Double.parseDouble(dl.getProperty("angle", "NaN")), cimNamespace, writer, context);
                } else if (b != null) {
                    writeVoltage(topologicalNode, dl.getBoundary().getV(), dl.getBoundary().getAngle(), cimNamespace, writer, context);
                } else {
                    writeVoltage(topologicalNode, 0.0, 0.0, cimNamespace, writer, context);
                }
            }
        }
    }
```
in powsybl-core/cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java

**Comment by Haigutus** [comment]
Fix, drop one of the boundary SV results in post processing

**Comment by mr0321** [comment]
It seems that this is an intentional behaviour: [If two dangling lines with the same Xnode are in the same geographical region in a CGMES file, they are not merged #2389](https://github.com/powsybl/powsybl-core/pull/2389)
